force-app/main/default/classes/SOQL.cls

```diff
@@ -130,9 +194,14 @@ public with sharing class SOQL {
      * @return A list of Lead records.
      * Update the corresponding test class and testGetLeadsBySourceOrTitle method to cover this method
      */
-    //[public | private ] [static] data_type method_name (input parameters) {
+    public static List<Lead> getLeadsBySourceOrTitle(String leadSource, String keyword) {
     //  The body of the method
-    //}    
+    if (!Schema.sObjectType.Lead.isAccessible()) {
+        System.debug('You do not have access to this Object.  Please contact your Salesforce Administrator.');
+    }
+
+    return [SELECT Id, LastName, Company, Phone, Email, LeadSource, Title FROM Lead WHERE LeadSource LIKE : '%' + leadSource + '%' OR Title LIKE : '%' + keyword + '%'];
```

Comment:
For the leadsource you want it to be equal instead of LIKE
Ex. if someone passed in 'Web' your results would return (Web, Website, SpiderWeb) if those values existed.
```suggestion
    return [SELECT Id, LastName, Company, Phone, Email, LeadSource, Title FROM Lead WHERE LeadSource =: leadSource OR Title LIKE : ('%' + keyword + '%')];
```